### PR TITLE
Add review approval comment basic structs and logic

### PR DESCRIPTION
**Pull request by tapishr**
Please check the action items covered in the PR - 

- [x] Build is running
- [ ] Eventing is functional and tested
- [ ] Unit or integration tests added and running
- [ ] Manual QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a new approval process for handling approval messages, including parsing data and calculating contributions.
	- Added functionality for handling git aliases, including retrieving and sending aliases for git repositories.
	- Implemented database operations for aliases, allowing saving, retrieving, and updating aliases.

- **Enhancements**
	- Improved the handling of private status in repositories to effectively address deserialization errors.
	- Streamlined GitHub setup operations by enhancing the handling of repository aliases and removing redundant operations.

- **Refactor**
	- Reordered imports and adjusted function signatures in the coverage module to refine logic and improve function calls.
	- Added new modules to the core and utils sections, expanding the codebase's functionality.

- **Bug Fixes**
	- Fixed an issue in the pull request processing logic by integrating the new approval process.

### CODE REVIEW

**Comment by MuskanPaliwal** [comment]
Relevant users for this PR:

| Contributor Name/Alias  | Relevance |
| -------------- | --------------- |
| muskan10112002@gmail.com | 22.22% |
| tapish303@gmail.com | 77.78% |


Auto assigning to all relevant reviewers

If you are a relevant reviewer, you can use the [Vibinex browser extension](https://chromewebstore.google.com/detail/vibinex-code-review/jafgelpkkkopeaefadkdjcmnicgpcncc) to see parts of the PR relevant to you
Relevance of the reviewer is calculated based on the git blame information of the PR. To know more, hit us up at contact@vibinex.com.

To change comment and auto-assign settings, go to [your Vibinex settings page.](https://vibinex.com/u)

**Comment by MuskanPaliwal** [comment]
Relevant users for this PR:

| Contributor Name/Alias  | Relevance |
| -------------- | --------------- |
| muskan10112002@gmail.com | 4.65% |
| tapish303@gmail.com | 95.35% |


Auto assigning to all relevant reviewers

If you are a relevant reviewer, you can use the [Vibinex browser extension](https://chromewebstore.google.com/detail/vibinex-code-review/jafgelpkkkopeaefadkdjcmnicgpcncc) to see parts of the PR relevant to you
Relevance of the reviewer is calculated based on the git blame information of the PR. To know more, hit us up at contact@vibinex.com.

To change comment and auto-assign settings, go to [your Vibinex settings page.](https://vibinex.com/u)

**Review by MuskanPaliwal — DISMISSED** [review]
For testing purpose only

File: vibi-dpu/src/db/review.rs [hunk 1]
```diff
@@ -15,4 +15,27 @@ pub fn save_review_to_db(review: &Review) {
...
+    let review_res = serde_json::from_slice(&ivec);
+    if let Err(e) = review_res {
+        log::error!(
+            "[get_handles_from_db] Failed to deserialize aliases from json: {:?}",
+            e
+        );
+        return None;
+    }
+    let review: Review = review_res.expect("Uncaught error in alias_res");
+    return Some(review);
```
[the remark this answers is not in the document]
Comment (tapishr) [reply vibi-dpu/src/db/review.rs:40]: done

**Pull request merged**